### ISSUE TITLE
Multiple loading of calendar multiply Header XIB file registered resulting crash due to "Identifier was not registered:" 

### ISSUE THREAD

**Issue by kristof4ad** [patchthecode/JTAppleCalendar#40]
Im using calendar to filter results by date, when calendar is displayed for the first time everything is OK but after dismissing view and trying load for a second time, method self.calendarView.registerHeaderViewXibs(fileNames: ["HeaderView"]) loading the same file for a second time and than condition  "if headerViewXibs.count == 1"  is not met resulting crash:
assertion failed: Identifier was not registered: 
![screen shot 2016-06-06 at 15 47 29](https://cloud.githubusercontent.com/assets/10072495/15826332/126b826a-2bff-11e6-88c7-e4e57d1cb607.png)
file.../Pods/JTAppleCalendar/Pod/Classes/JTAppleCalendarDelegates.swift, line 154
I'm using only one xib file for HeaderTemplate

**Comment by patchthecode** [comment]
Thanks for letting me know. One thing though. I am not sure how you are reloading your view etc. So can you let me know the following?

1. Which version are you using?
2. if you are using version 4.0.0, then can you do the following and let me know if it works?

In the `UserInteractionFunctions.swift` file, can you change the first few lines of code from this
```swift
    public func registerHeaderViewXibs(fileNames headerViewXibNames: [String]) {
        if headerViewXibNames.count < 1 {
            headerViewXibs.removeAll()
            return
        }
```
To this?
```swift
    public func registerHeaderViewXibs(fileNames headerViewXibNames: [String]) {
        headerViewXibs.removeAll()
        if headerViewXibNames.count < 1 {
            return
        }
```
And let me know if you are still experiencing the issue? If you still are having the problem I have to look into it deeper. If not, then I will do a patch.

**Comment by kristof4ad** [comment]
Hi, 
Thanks for a quick replay!
I'm using version 3.0.1 which is the latest on CocoaPods
I guess 4.0.0 is in beta... 
Yes "headerViewXibs.removeAll()" is the kind of method I was looking for.
Unfortunately on 3.0.1 it do not work.
![screen shot 2016-06-07 at 09 20 12](https://cloud.githubusercontent.com/assets/10072495/15850749/a1277896-2c91-11e6-87ba-ae9f18a37254.png)

**Comment by kristof4ad** [comment]
I found in version 3.0.1 registerHeaderViewXibs method. 
I try to modify It but the code is locked.
I'm not sure if it safe for me to make fixes this way  because future updates might revert them.

Regarding reloading me view, when somebody pick a date I'm saving it and dismiss View Controller.
If they want to pick another day I'm loading calendar page again.
Regards 
Kristof

**Comment by patchthecode** [comment]
4.0.0 is in beta? Nah mate :) 
I move very fast in the fixing of bugs etc. `4.0.0` is the latest version.
Very many bugs have been fixed in that version, including:

1. Performance issues
2. crashes etc

I just checked cocoapods and its showing version 4.0.0 as latest. Can you try changing this to version 4.0.0 in your podspec file?

Also note, in version 4.0, the way you configure the calendar has been changed slightly. Everything else has remained the same.

**Comment by kristof4ad** [comment]
OK I will update to 4.0.0
Thanks

**Comment by patchthecode** [comment]
And btw, in regards to the unlocking of the cocopod, there is no problem there.
Just unlock it and do the code change to test if it works.
If it works, then i know exactly what the problem is. And i'll release a 4.0.1 patch to fix it. 
Unlocking the cocoapod and changing it will get overridden when you update to version 4.0.1.

It will not mess up and thing so no worries.

**Comment by kristof4ad** [comment]
Yes, updating to 4.0.0 and moving  headerViewXibs.removeAll() outside if statement solved the problem.
Thank you

**Comment by patchthecode** [comment]
Ok. I will add the fix to a version 4.0.1 patch and release the fix over the weekend. thanks for helping me solve this.
